modules/git/repo_compare.go

```diff
@@ -263,3 +270,42 @@ func (repo *Repository) GetDiffFromMergeBase(base, head string, w io.Writer) err
 	}
 	return err
 }
+
+// ReadPatchCommit will check if a diff patch exists and return stats
+func (repo *Repository) ReadPatchCommit(prID int64) (commitSHA string, err error) {
+	// Migrated repositories download patches to "pulls" location
+	patchFile := fmt.Sprintf("pulls/%d.patch", prID)
+	loadPatch, err := os.Open(filepath.Join(repo.Path, patchFile))
+	if err != nil {
+		return "", err
+	}
+	defer loadPatch.Close()
+	// Read only the first line of the patch - usually it contains the first commit made in patch
+	scanner := bufio.NewScanner(loadPatch)
+	scanner.Scan()
+	// Parse the Patch stats, sometimes Migration returns a 404 for the patch file
+	commitSHAGroups := patchCommits.FindStringSubmatch(scanner.Text())
+	if len(commitSHAGroups) != 0 {
+		commitSHA = commitSHAGroups[1]
+	} else {
+		return "", errors.New("patch file unreadable")
+	}
+	return commitSHA, nil
+}
+
+// WritePullHead will populate a PR head retrieved from patch file
+func (repo *Repository) WritePullHead(prID int64, commitSHA string) error {
+	headPath := fmt.Sprintf("refs/pull/%d", prID)
+	fullHeadPath := filepath.Join(repo.Path, headPath)
+	// Create missing directory just in case
+	if err := os.MkdirAll(fullHeadPath, os.ModePerm); err != nil {
+		return err
+	}
+	commitBytes := []byte(commitSHA)
+	pullPath := filepath.Join(fullHeadPath, "head")
+	err := ioutil.WriteFile(pullPath, commitBytes, os.ModePerm)
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
perhaps:

```suggestion
	return ioutil.WriteFile(pullPath, commitBytes, os.ModePerm)
```

Reply:
Thank you it is indeed easier to read and better !